data/trie/trieStorageManager.go

```diff
@@ -435,16 +423,29 @@ func (tsm *trieStorageManager) getSnapshotDb(newDb bool) data.DBWriteCacher {
 func (tsm *trieStorageManager) removeSnapshot() {
 	dbUniqueId := strconv.Itoa(tsm.snapshotId - len(tsm.snapshots))
```

Comment:
Could be called with 0 snapshots? If yes, I think you should check if len(tsm.snapshots) > 0, and if not, you can return in first line, otherwise I think it could panic in line 427.

Reply:
As the code is written now, this can not be called with 0 snapshots. Added the check anyway, just to be safe in the future if the code changes.